assembl/views/api2/auth.py

```diff
@@ -354,9 +362,15 @@ def reset_password(request):
     elif identifier:
         user, account = from_identifier(identifier)
         if not user:
-            raise JSONError(
-                localizer.translate(_("This email does not exist")),
-                code=HTTPNotFound.code)
+            if not discussion.preferences['generic_auth_errors']:
+                raise JSONError(
+                    localizer.translate(_("This email does not exist")),
+                    code=HTTPNotFound.code)
+            else:
+                raise JSONError(
+                    localizer.translate(_(generic_error_message)),
+                    code=HTTPNotFound.code)
+                logger.error("This email does not exist.")
```

Comment:
Again, follow the above example to be more specific. In a long logger, need to search for this quickly, and in a lot of other information, need clear information of which view the error came from, and what the trigger point was.